libsolidity/formal/CHC.cpp

```diff
@@ -1001,6 +1015,37 @@ set<unsigned> CHC::transactionVerificationTargetsIds(ASTNode const* _txRoot)
 	return verificationTargetsIds;
 }
 
+optional<CHC::CHCNatspecOption> CHC::natspecOptionFromString(string const& _option)
+{
+	static map<string, CHCNatspecOption> options{
+		{"abstract-function-nondet", CHCNatspecOption::AbstractFunctionNondet}
+	};
+	if (options.count(_option))
+		return options.at(_option);
+	return {};
+}
+
+set<CHC::CHCNatspecOption> CHC::smtNatspecTags(FunctionDefinition const& _function)
+{
+	set<CHC::CHCNatspecOption> options;
+	auto const& tags = _function.annotation().docTags;
+	string smtStr = "custom:smtchecker";
+	for (auto it = tags.find(smtStr); it != tags.end() && it->first == smtStr; ++it)
+	{
+		string const& content = it->second.content;
+		if (auto option = natspecOptionFromString(content))
```

Comment:
Would this work for `/// @custom:smtchecker  abstract-function-nondet`, i.e., an additional space before `abstract-...`.

Reply:
Not sure, that depends how natspec parses it.